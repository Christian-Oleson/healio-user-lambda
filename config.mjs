export const APP_NAME = 'Healiospace';

