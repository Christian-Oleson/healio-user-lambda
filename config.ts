export const APP_NAME: string = "Healiospace";
