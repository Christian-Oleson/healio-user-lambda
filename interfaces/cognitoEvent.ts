export interface CognitoEvent {
  triggerSource: string;
  request: {
    codeParameter: string;
    linkParameter?: string;
    usernameParameter?: string;
    userAttributes?: {
      email: string;
      family_name?: string;
      given_name?: string | null;
      [key: string]: string | undefined | null;
    }
  };
  response: { [key: string]: any };
}
