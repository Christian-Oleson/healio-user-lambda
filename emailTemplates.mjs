import { APP_NAME } from './config.mjs';

const baseStyles = `
  body {
    font-family: 'Segoe UI', 'Roboto', Arial, sans-serif;
    background: #f7f8fa;
    color: #222;
    margin: 0;
    padding: 0;
  }
  .container {
    background: #fff;
    max-width: 480px;
    margin: 40px auto;
    border-radius: 10px;
    box-shadow: 0 2px 8px rgba(0,0,0,0.07);
    padding: 32px 28px 24px 28px;
  }
  .brand {
    color: #4f46e5;
    font-weight: 700;
    font-size: 1.5em;
    margin-bottom: 8px;
    letter-spacing: 1px;
  }
  .code {
    display: inline-block;
    background: #f1f5f9;
    color: #1e293b;
    font-size: 1.2em;
    font-weight: bold;
    padding: 8px 18px;
    border-radius: 6px;
    margin: 16px 0;
    letter-spacing: 2px;
  }
  .footer {
    margin-top: 32px;
    font-size: 0.95em;
    color: #64748b;
    text-align: center;
  }
`;

export function signUpEmail(code) {
  return `<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome to ${APP_NAME}</title>
    <style>${baseStyles}</style>
  </head>
  <body>
    <div class="container">
      <div class="brand">${APP_NAME}</div>
      <h2>Welcome!</h2>
      <p>Thank you for joining <b>${APP_NAME}</b>. To complete your registration, please use the verification code below:</p>
      <div class="code">${code}</div>
      <p>If you did not request this, you can safely ignore this email.</p>
      <div class="footer">&copy; ${new Date().getFullYear()} ${APP_NAME}. All rights reserved.</div>
    </div>
  </body>
</html>`;
}

export function forgotPasswordEmail(code) {
  return `<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Reset Your ${APP_NAME} Password</title>
    <style>${baseStyles}</style>
  </head>
  <body>
    <div class="container">
      <div class="brand">${APP_NAME}</div>
      <h2>Password Reset Request</h2>
      <p>We received a request to reset your <b>${APP_NAME}</b> password. Use the code below to proceed:</p>
      <div class="code">${code}</div>
      <p>If you did not request this, you can safely ignore this email.</p>
      <div class="footer">&copy; ${new Date().getFullYear()} ${APP_NAME}. All rights reserved.</div>
    </div>
  </body>
</html>`;
}

export function adminCreateUserEmail(code) {
  return `<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Your ${APP_NAME} Account</title>
    <style>${baseStyles}</style>
  </head>
  <body>
    <div class="container">
      <div class="brand">${APP_NAME}</div>
      <h2>Your Account Has Been Created</h2>
      <p>An administrator has created an account for you on <b>${APP_NAME}</b>. Use the temporary password below to sign in and set your own password:</p>
      <div class="code">${code}</div>
      <p>Please sign in and change your password as soon as possible.</p>
      <div class="footer">&copy; ${new Date().getFullYear()} ${APP_NAME}. All rights reserved.</div>
    </div>
  </body>
</html>`;
}

