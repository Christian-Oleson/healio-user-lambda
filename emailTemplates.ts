import { APP_NAME } from "./config.js";
import { CognitoEvent } from "./interfaces/cognitoEvent.js";

const rootURL = "https://main.d2yakdlwmmo9ua.amplifyapp.com";
const baseStyles = `
    body {
        font-family: 'Segoe UI', 'Roboto', Arial, sans-serif;
        background: #f7f8fa;
        color: #222;
        margin: 0;
        padding: 0;
        line-height: 1.5;
    }
    .container {
        background: #fff;
        max-width: 580px;
        margin: 40px auto;
        border-radius: 10px;
        box-shadow: 0 4px 12px rgba(0,0,0,0.1);
        padding: 40px 32px;
    }
    .header {
        text-align: center;
        margin-bottom: 25px;
    }
    .brand {
        color: #4f46e5;
        font-weight: 700;
        font-size: 2em;
        margin-bottom: 8px;
        letter-spacing: 1px;
    }
    h2 {
        color: #1e293b;
        font-weight: 600;
        font-size: 1.5rem;
        margin-top: 0;
        margin-bottom: 20px;
    }
    .code-container {
        text-align: center;
        margin: 30px 0;
    }
    .code {
        display: inline-block;
        background: #f1f5f9;
        color: #1e293b;
        font-size: 1.4em;
        font-weight: bold;
        padding: 12px 24px;
        border-radius: 6px;
        letter-spacing: 2px;
        text-align: center;
    }
    .button {
        display: inline-block;
        background-color: #4f46e5;
        color: white !important;
        text-decoration: none;
        padding: 12px 24px;
        border-radius: 6px;
        font-weight: 600;
        margin: 24px 0;
        text-align: center;
    }
    .secondary-text {
        color: #64748b;
        font-size: 0.95em;
        margin-top: 16px;
    }
    .divider {
        border-top: 1px solid #e2e8f0;
        margin: 32px 0;
    }
    .footer {
        margin-top: 32px;
        font-size: 0.9em;
        color: #64748b;
        text-align: center;
    }
`;

function baseEmailTemplate({
  title,
  heading,
  message,
  codeLabel,
  code,
  buttonText,
  buttonUrl,
  secondaryText,
  showButton = true,
}: {
  title: string;
  heading: string;
  message: string;
  codeLabel?: string;
  code?: string;
  buttonText?: string;
  buttonUrl?: string;
  secondaryText?: string;
  showButton?: boolean;
}): string {
  const currentYear = new Date().getFullYear();
  return `<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>${title}</title>
    <style>${baseStyles}</style>
</head>
<body>
<div class="container">
    <div class="header">
        <div class="brand">${APP_NAME}</div>
    </div>
    <h2>${heading}</h2>
    <p>${message}</p>
    ${
      code
        ? `<div class="code-container">
        ${codeLabel ? `<p><strong>${codeLabel}</strong></p>` : ""}
        <div class="code">${code}</div>
    </div>`
        : ""
    }
    ${showButton && buttonText && buttonUrl ? `<p style="text-align: center;"><a href="${buttonUrl}" class="button">${buttonText}</a></p>` : ""}
    ${secondaryText ? `<p class="secondary-text">${secondaryText}</p>` : ""}
    <div class="divider"></div>
    <div class="footer">
        <p>Â© ${currentYear} ${APP_NAME}. All rights reserved.</p>
        <p>Powered by HealioSpace - Transform Your Health Journey</p>
    </div>
</div>
</body>
</html>`;
}

export function signUpEmail(code: string, showButton: boolean = false): string {
  return baseEmailTemplate({
    title: `Welcome to ${APP_NAME}`,
    heading: `Welcome to ${APP_NAME}!`,
    message: `Thank you for joining <b>${APP_NAME}</b>. To complete your registration, please use the verification code below:`,
    codeLabel: "Your Verification Code:",
    code,
    buttonText: "Sign In Now",
    buttonUrl: `${rootURL}/login`,
    secondaryText:
      "If you did not request this, you can safely ignore this email.",
    showButton,
  });
}

export function forgotPasswordEmail(cognitoEvent: CognitoEvent): string {
  return baseEmailTemplate({
    title: `Reset Your ${APP_NAME} Password`,
    heading: "Password Reset Request",
    message: `We received a request to reset your <b>${APP_NAME}</b> password. Use the code below to proceed:`,
    codeLabel: "Your Password Reset Code:",
    code: cognitoEvent.request.codeParameter,
    buttonText: "Reset Password",
    buttonUrl: `${rootURL}/login`,
    secondaryText:
      "If you did not request this, you can safely ignore this email.",
    showButton: false,
  });
}

export function adminCreateUserEmail(
  code: string,
  username: string = "",
  showButton: boolean = true,
): string {
  return baseEmailTemplate({
    title: `Welcome to ${APP_NAME}`,
    heading: `Welcome to ${APP_NAME}!`,
    message: `You have been invited to join the ${APP_NAME} platform. We're excited to have you on board!<br><br><strong>Getting Started:</strong><ol><li>Use the temporary password below to sign in</li><li>You'll be prompted to create your own password</li><li>Complete your profile to get the most out of ${APP_NAME}</li></ol>`,
    codeLabel: "Your Temporary Password:",
    code,
    buttonText: "Sign In Now",
    buttonUrl: `${rootURL}/login`,
    secondaryText:
      'If you have any questions or need assistance, please contact our support team at <a href="mailto:support@dosapes.com">support@dosapes.com</a>.',
    showButton,
  });
}

export function preSignUpEmail(
  givenName: string = "User",
  code?: string,
  showButton: boolean = false,
): string {
  return baseEmailTemplate({
    title: `Welcome to ${APP_NAME}`,
    heading: `Welcome, ${givenName}!`,
    message: `Thank you for signing up for <b>${APP_NAME}</b>. We're excited to have you join our community!<br><br>Your account is being processed${code ? ", and you can use the verification code below to complete your registration." : ", and you will receive a verification code shortly to complete your registration."}`,
    codeLabel: code ? "Your verification code:" : undefined,
    code,
    buttonText: "Learn More",
    buttonUrl: `${rootURL}/about`,
    secondaryText:
      'If you have any questions or need assistance, please contact our support team at <a href="mailto:support@dosapes.com">support@dosapes.com</a>.',
    showButton,
  });
}
