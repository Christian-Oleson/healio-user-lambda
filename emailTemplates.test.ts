import { describe, it, expect } from 'vitest';
import { signUpEmail, forgotPasswordEmail, adminCreateUserEmail } from './emailTemplates.js';
import { handler } from './index.js';
import { APP_NAME } from './config.js';
import { CognitoEvent } from "./interfaces/cognitoEvent.js";

// Email template tests

describe('Email Templates', () => {
  it('signUpEmail includes code, app name, and lang attribute', () => {
    const code = '123456';
    const html = signUpEmail(code);
    expect(html).toContain(code);
    expect(html).toContain(APP_NAME);
    expect(html).toContain('<html lang="en">');
  });

  it('forgotPasswordEmail includes code and app name', () => {
    const code = '654321';
    const cognitoEvent: CognitoEvent = {
        triggerSource: 'CustomMessage_ForgotPassword',
        request: { codeParameter: code },
        response: {}
    }
    const html = forgotPasswordEmail(cognitoEvent);
    expect(html).toContain(code);
    expect(html).toContain(APP_NAME);
  });

  it('adminCreateUserEmail includes code and app name', () => {
    const code = 'abcdef';
    const html = adminCreateUserEmail(code);
    expect(html).toContain(code);
    expect(html).toContain(APP_NAME);
  });
});

describe('Email Templates HTML', () => {
  it('signUpEmail HTML structure and content', () => {
    const code = 'abc123';
    const html = signUpEmail(code);

    // Expect
    expect(html).toMatchSnapshot();
  });

  it('forgotPasswordEmail HTML structure and content', () => {
    const code = 'reset321';
    const cognitoEvent: CognitoEvent = {
      triggerSource: 'CustomMessage_ForgotPassword',
      request: { codeParameter: code },
      response: {}
    }
    const html = forgotPasswordEmail(cognitoEvent);

    // Expect
    expect(html).toMatchSnapshot();
  });

  it('adminCreateUserEmail HTML structure and content', () => {
    const code = 'tempPass!';
    const html = adminCreateUserEmail(code);

    // Expect
    expect(html).toMatchSnapshot();
  });
});

// Handler tests

describe('Lambda Handler', () => {
  it('handles CustomMessage_SignUp', async () => {
    const event: CognitoEvent = {
      triggerSource: 'CustomMessage_SignUp',
      request: { codeParameter: '111111' },
      response: {}
    };
    const result = await handler(event);
    expect(result.response.emailMessage).toContain('111111');
    expect(result.response.emailSubject).toContain(APP_NAME);
  });

  it('handles CustomMessage_ForgotPassword', async () => {
    const event: CognitoEvent = {
      triggerSource: 'CustomMessage_ForgotPassword',
      request: { codeParameter: '222222' },
      response: {}
    };
    const result = await handler(event);
    expect(result.response.emailMessage).toContain('222222');
  });

  it('handles CustomMessage_AdminCreateUser', async () => {
    const event: CognitoEvent = {
      triggerSource: 'CustomMessage_AdminCreateUser',
      request: { codeParameter: '333333' },
      response: {}
    };
    const result = await handler(event);
    expect(result.response.emailMessage).toContain('333333');
  });
});
