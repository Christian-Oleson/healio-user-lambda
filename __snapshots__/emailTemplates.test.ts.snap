// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Email Templates HTML > adminCreateUserEmail HTML structure and content 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome to Healiospace</title>
    <style>
    body {
        font-family: 'Segoe UI', 'Roboto', Arial, sans-serif;
        background: #f7f8fa;
        color: #222;
        margin: 0;
        padding: 0;
        line-height: 1.5;
    }
    .container {
        background: #fff;
        max-width: 580px;
        margin: 40px auto;
        border-radius: 10px;
        box-shadow: 0 4px 12px rgba(0,0,0,0.1);
        padding: 40px 32px;
    }
    .header {
        text-align: center;
        margin-bottom: 25px;
    }
    .brand {
        color: #4f46e5;
        font-weight: 700;
        font-size: 2em;
        margin-bottom: 8px;
        letter-spacing: 1px;
    }
    h2 {
        color: #1e293b;
        font-weight: 600;
        font-size: 1.5rem;
        margin-top: 0;
        margin-bottom: 20px;
    }
    .code-container {
        text-align: center;
        margin: 30px 0;
    }
    .code {
        display: inline-block;
        background: #f1f5f9;
        color: #1e293b;
        font-size: 1.4em;
        font-weight: bold;
        padding: 12px 24px;
        border-radius: 6px;
        letter-spacing: 2px;
        text-align: center;
    }
    .button {
        display: inline-block;
        background-color: #4f46e5;
        color: white !important;
        text-decoration: none;
        padding: 12px 24px;
        border-radius: 6px;
        font-weight: 600;
        margin: 24px 0;
        text-align: center;
    }
    .secondary-text {
        color: #64748b;
        font-size: 0.95em;
        margin-top: 16px;
    }
    .divider {
        border-top: 1px solid #e2e8f0;
        margin: 32px 0;
    }
    .footer {
        margin-top: 32px;
        font-size: 0.9em;
        color: #64748b;
        text-align: center;
    }
</style>
</head>
<body>
<div class="container">
    <div class="header">
        <div class="brand">Healiospace</div>
    </div>
    <h2>Welcome to Healiospace!</h2>
    <p>You have been invited to join the Healiospace platform. We're excited to have you on board!<br><br><strong>Getting Started:</strong><ol><li>Use the temporary password below to sign in</li><li>You'll be prompted to create your own password</li><li>Complete your profile to get the most out of Healiospace</li></ol></p>
    <div class="code-container">
        <p><strong>Your Temporary Password:</strong></p>
        <div class="code">tempPass!</div>
    </div>
    <p style="text-align: center;"><a href="https://main.d2yakdlwmmo9ua.amplifyapp.com//login" class="button">Sign In Now</a></p>
    <p class="secondary-text">If you have any questions or need assistance, please contact our support team at <a href="mailto:support@dosapes.com">support@dosapes.com</a>.</p>
    <div class="divider"></div>
    <div class="footer">
        <p>© 2025 Healiospace. All rights reserved.</p>
        <p>Powered by HealioSpace - Transform Your Health Journey</p>
    </div>
</div>
</body>
</html>"
`;

exports[`Email Templates HTML > forgotPasswordEmail HTML structure and content 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Reset Your Healiospace Password</title>
    <style>
    body {
        font-family: 'Segoe UI', 'Roboto', Arial, sans-serif;
        background: #f7f8fa;
        color: #222;
        margin: 0;
        padding: 0;
        line-height: 1.5;
    }
    .container {
        background: #fff;
        max-width: 580px;
        margin: 40px auto;
        border-radius: 10px;
        box-shadow: 0 4px 12px rgba(0,0,0,0.1);
        padding: 40px 32px;
    }
    .header {
        text-align: center;
        margin-bottom: 25px;
    }
    .brand {
        color: #4f46e5;
        font-weight: 700;
        font-size: 2em;
        margin-bottom: 8px;
        letter-spacing: 1px;
    }
    h2 {
        color: #1e293b;
        font-weight: 600;
        font-size: 1.5rem;
        margin-top: 0;
        margin-bottom: 20px;
    }
    .code-container {
        text-align: center;
        margin: 30px 0;
    }
    .code {
        display: inline-block;
        background: #f1f5f9;
        color: #1e293b;
        font-size: 1.4em;
        font-weight: bold;
        padding: 12px 24px;
        border-radius: 6px;
        letter-spacing: 2px;
        text-align: center;
    }
    .button {
        display: inline-block;
        background-color: #4f46e5;
        color: white !important;
        text-decoration: none;
        padding: 12px 24px;
        border-radius: 6px;
        font-weight: 600;
        margin: 24px 0;
        text-align: center;
    }
    .secondary-text {
        color: #64748b;
        font-size: 0.95em;
        margin-top: 16px;
    }
    .divider {
        border-top: 1px solid #e2e8f0;
        margin: 32px 0;
    }
    .footer {
        margin-top: 32px;
        font-size: 0.9em;
        color: #64748b;
        text-align: center;
    }
</style>
</head>
<body>
<div class="container">
    <div class="header">
        <div class="brand">Healiospace</div>
    </div>
    <h2>Password Reset Request</h2>
    <p>We received a request to reset your <b>Healiospace</b> password. Use the code below to proceed:</p>
    <div class="code-container">
        <p><strong>Your Password Reset Code:</strong></p>
        <div class="code">reset321</div>
    </div>
    <p style="text-align: center;"><a href="https://main.d2yakdlwmmo9ua.amplifyapp.com//login" class="button">Reset Password</a></p>
    <p class="secondary-text">If you did not request this, you can safely ignore this email.</p>
    <div class="divider"></div>
    <div class="footer">
        <p>© 2025 Healiospace. All rights reserved.</p>
        <p>Powered by HealioSpace - Transform Your Health Journey</p>
    </div>
</div>
</body>
</html>"
`;

exports[`Email Templates HTML > signUpEmail HTML structure and content 1`] = `
"<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome to Healiospace</title>
    <style>
    body {
        font-family: 'Segoe UI', 'Roboto', Arial, sans-serif;
        background: #f7f8fa;
        color: #222;
        margin: 0;
        padding: 0;
        line-height: 1.5;
    }
    .container {
        background: #fff;
        max-width: 580px;
        margin: 40px auto;
        border-radius: 10px;
        box-shadow: 0 4px 12px rgba(0,0,0,0.1);
        padding: 40px 32px;
    }
    .header {
        text-align: center;
        margin-bottom: 25px;
    }
    .brand {
        color: #4f46e5;
        font-weight: 700;
        font-size: 2em;
        margin-bottom: 8px;
        letter-spacing: 1px;
    }
    h2 {
        color: #1e293b;
        font-weight: 600;
        font-size: 1.5rem;
        margin-top: 0;
        margin-bottom: 20px;
    }
    .code-container {
        text-align: center;
        margin: 30px 0;
    }
    .code {
        display: inline-block;
        background: #f1f5f9;
        color: #1e293b;
        font-size: 1.4em;
        font-weight: bold;
        padding: 12px 24px;
        border-radius: 6px;
        letter-spacing: 2px;
        text-align: center;
    }
    .button {
        display: inline-block;
        background-color: #4f46e5;
        color: white !important;
        text-decoration: none;
        padding: 12px 24px;
        border-radius: 6px;
        font-weight: 600;
        margin: 24px 0;
        text-align: center;
    }
    .secondary-text {
        color: #64748b;
        font-size: 0.95em;
        margin-top: 16px;
    }
    .divider {
        border-top: 1px solid #e2e8f0;
        margin: 32px 0;
    }
    .footer {
        margin-top: 32px;
        font-size: 0.9em;
        color: #64748b;
        text-align: center;
    }
</style>
</head>
<body>
<div class="container">
    <div class="header">
        <div class="brand">Healiospace</div>
    </div>
    <h2>Welcome to Healiospace!</h2>
    <p>Thank you for joining <b>Healiospace</b>. To complete your registration, please use the verification code below:</p>
    <div class="code-container">
        <p><strong>Your Verification Code:</strong></p>
        <div class="code">abc123</div>
    </div>
    <p style="text-align: center;"><a href="https://main.d2yakdlwmmo9ua.amplifyapp.com//login" class="button">Sign In Now</a></p>
    <p class="secondary-text">If you did not request this, you can safely ignore this email.</p>
    <div class="divider"></div>
    <div class="footer">
        <p>© 2025 Healiospace. All rights reserved.</p>
        <p>Powered by HealioSpace - Transform Your Health Journey</p>
    </div>
</div>
</body>
</html>"
`;
